hibernate-core/src/main/java/org/hibernate/engine/query/spi/EntityGraphQueryHint.java

```diff
@@ -143,7 +141,7 @@ else if ( propertyType.isCollectionType() ) {
 
 						final FromElementFactory fromElementFactory = new FromElementFactory(
 								fromClause, origin,
-								attributeName, classAlias, columns, false
+								attributeName, null, columns, false
```

Comment:
Obviously, we should not use `classAlias` here for it is the joining lhs and using it only ends up with screwing up internal `formElementByClassAlias` map in `FromClause` by overwriting lhs by a collection from element without persister, thus the NPE bug when `QueryLoader#applyLocks()` is invoked.